platform-services.md

```diff
@@ -0,0 +1,215 @@
+`FlutterView` interface. The `FlutterView` interface allows you to embed a bi-directional
+asynchronous messaging component in your platform code to send and receive native service calls
+in a transparent way. This flexible approach lets you create cross-platform mobile apps that can
+connect to platform-specific and third-party services on iOS and Android devices, while sharing a
+common user interface (UI) codebase in Dart.
+
+You can build your own integrations with platform-specific services and third-party SDKs. Ideally,
+the services that you invoke using the `FlutterView` interface should return non-graphical,
+data-only information. Examples include:
+
+*   Geolocation
+*   Contacts
+*   File system
+*   Network information
+*   Sensors
+
+You can also use the `FlutterView` interface to invoke APIs from third-party SDKs that run on iOS
+and Android devices.
+
+To use the `FlutterView` interface, you must add a platform-specific Flutter component into the
+platform’s view hierarchy. For example, on iOS, you add a `FlutterViewController` object to
+interact with a `UIView`.  Similarly, on Android, you add a `FlutterView` object to your app’s
+`android.view.View` hierarchy. Your app can then communicate between the Dart and iOS or
+Android-specific parts of the app by using a simple message-passing interface.
+
+### Message passing {#message-passing}
+
+Flutter offers a simple message-passing interface between your platform-specific code and the
+Dart code. The message passing interface is asynchronous, bi-directional, and supports optional
+reply routing. Messages consist of strings encoded in JSON format, and these is no facility for
+transferring handles to other objects along with messages. You are responsible for encoding and
+decoding messages sent through this mechanism.
+
+## Get the `hello_services` sample {#the-hello_services-sample}
+
+The Flutter SDK includes a reference sample project called
+[`hello_services`](https://github.com/flutter/flutter/tree/master/examples/hello_services)
+that demonstrates how to connect to the geolocation service within an iOS or Android application
+by using the `FlutterView` interface.
+
+The `hello_services` project comes with helper code to handle the passing of
+JSON-formatted messages to invoke platform services. If your app needs access to functionality
+provided by platform-specific iOS or Android services, we strongly encourage you to
+[download](https://github.com/flutter/flutter/tree/master/examples/hello_services) and use this
+sample project as the starter code to build your own Flutter app.
+
+The following diagram shows how the `hello_services` app uses the `FlutterView` interface to
+send and receive messages to platform and third-party services on iOS and Android.
+
+ ![hello_services architecture](/images/hello_services.png)
+
+**Fig 1.** Architecture of the `hello_services` sample.
+
+The Dart, iOS, and Android-specific components are related as follows:
```

Comment:
Again, "Dart" isn't parallel with "iOS" and "Android".  The languages aren't as important, so I'd replace "Dart" with something descriptive about the function of that part of your app instead.